framework/src/main/java/org/checkerframework/framework/type/TypeFromMemberVisitor.java

```diff
@@ -30,11 +31,17 @@ public AnnotatedTypeMirror visitVariable(VariableTree node, AnnotatedTypeFactory
         AnnotatedTypeMirror result = TypeFromTree.fromTypeTree(f, node.getType());
 
         // Add primary annotations
-        List<? extends AnnotationTree> annos = node.getModifiers().getAnnotations();
-        if (annos != null && !annos.isEmpty()) {
-            List<AnnotationMirror> ams = TreeUtils.annotationsFromTypeAnnotationTrees(annos);
+        List<? extends AnnotationTree> annoTrees = node.getModifiers().getAnnotations();
```

Comment:
This change only overrides `visitVariable`, wheras `addAnnotationsFromElement` is used in other contexts as well.
In particular, this probably doesn't do the right thing for method return types - they are not variables. Could you add a test to make sure?

Reply:
There is a test, `foo2`.  Because annotations from methods are decoded using the element rather than the tree, this logic isn't needed there. (I added a comment in that method below.) All other places where a declaration annotation can be written are variables.  So this should be enough.

Reply:
Sounds good.